proto/zds.proto

```diff
@@ -15,10 +15,17 @@ message ZdsHello {
   Version version = 1;
 }
 
+message PodInfo {
+  string name = 1;
+  string namespace = 2;
+  string uid = 4;
+}
+
 // Add a workload to the ztunnel. this will be accompanied by ancillary data contianing
 // the workload's netns file descriptor.
 message AddWorkload {
   string uid = 1;
```

Comment:
Still don't quite understand why we are passing down 2 UIDs to ztunnel for a pod add, when the one for ambient index can be computed from clusterID/name/namespace without this?

I'm fine with 2 (clearly, differently-named, commented) UIDs, where the one used for index lookup is computed internally by ztunnel on AddWorkload.

CNI agent is dealing with Actual Pods and the K8S API, and should probably traffic exclusively in K8S Pod UIDs, since it is conceivable (but not likely) that at that level we could get something in the same cluster/name/namespace but a different K8S UID, and mess ourselves up.

Reply:
Are you referring to the `string uid = 4` ? Yes, this can be removed. 

I think the broader question is that we need to decide if computing the uid that way (clusterID/name/namespace) is acceptable, as it creates an implicit dependency between the ztunnel an ambient index code

The other options being to ambient index to pass-in pod uuids, or change the node agent to send ambient-index style uuids.

Reply:
> Are you referring to the string uid = 4 ? Yes, this can be removed.

yes, just that was confusing me, aside from the broader questions - ty!